### DISCUSSION TITLE
`let` implementation using dict

### DISCUSSION THREAD

**Discussion by scauligi** [hylang/hy#2119] [General]
Is there a reason `let` stores things into a dict instead of using symbols directly?
E.g., why does this:
```hy
(let [x 4  y 3] (+ x y))
```
get compiled to this:
```python
_hyx_letXUffffX10 = {}
_hyx_letXUffffX10['x'] = 4
_hyx_letXUffffX10['y'] = 3
_hyx_letXUffffX10['x'] + _hyx_letXUffffX10['y']
```
instead of something like this?
```python
_hyx_letXUffffX10__x = 4
_hyx_letXUffffX10__y = 3
_hyx_letXUffffX10__x + _hyx_letXUffffX10__y
```

**Comment by Kodiologist** [comment]
Because we wanted the assignment semantics of indexed assignment. There's no need for `nonlocal`, for example.
